src/Files.App/Utils/Git/GitHelpers.cs

```diff
@@ -646,11 +709,17 @@ public static bool IsRepositoryEx(string path, out string repoRootPath)
 			if (string.IsNullOrEmpty(repositoryRootPath))
 				return false;
 
-			if (Repository.IsValid(repositoryRootPath))
+			// Workaround for https://github.com/libgit2/libgit2sharp/issues/1851
+
+			try
 			{
-				repoRootPath = repositoryRootPath;
-				return true;
+				if (Repository.IsValid(repositoryRootPath))
+				{
+					repoRootPath = repositoryRootPath;
```

Comment:
I would use `SafetyExtensions.IgnoreExceptions` here but it doesn't allow access to `repoRootPath`.